### PR TITLE
Create dot_marketplace_v2-milestone_1.md

### PR DESCRIPTION
**Pull request by varun24143**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/832

### CODE REVIEW

File: deliveries/dot_marketplace_v2-milestone_1.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Milestone Delivery :mailbox:
+
+**The [invoice form :pencil:](https://docs.google.com/forms/d/e/1FAIpQLSdSqj2vYjvpiIytkjcc40Pwl0Eg76WGUAq5L9e8eFuuOegmLw/viewform) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/General-Grants-Program/blob/master/grants/milestone-deliverables-guidelines.md).**
+
+* **PR Link:** https://github.com/w3f/Grants-Program/pull/832
```
Comment (semuelle) [inline deliveries/dot_marketplace_v2-milestone_1.md:5]:
```suggestion
* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/dot_marketplace-phase2.md
```

**Review by semuelle — CHANGES_REQUESTED** [review]

**Comment by takahser** [comment]
@varun24143 Thanks for your patience and sorry for the delay here, we have currently a high number of open deliveries that we have to evaluate.
I'm going to look into this as soon as I can, hopefully tomorrow. 👍

**Comment by takahser** [comment]
@varun24143 I'm a bit confused why I have to

**Comment by varun24143** [comment]
> @varun24143 I'm a bit confused why I have to [clone](https://github.com/WowLabz/dot-marketplace-v2/blob/Phase2_Milestone1/testing_guide_phase2_milestone1.md#docker-guide) the seemingly [old repo](https://github.com/WowLabz/dot_marketplace_docker/tree/Phase2_Milestone1). It seems to me that the [dot-marketplace-v2](https://github.com/WowLabz/dot-marketplace-v2) is the repo for phase 2 while [dot_marketplace_docker](https://github.com/WowLabz/dot_marketplace_docker) is the repo for phase 1 of DOT Marketplace, is that correct?

> @takahser In the same repo we have another branch which is the [phase2_milestone1](https://github.com/WowLabz/dot_marketplace_docker/tree/Phase2_Milestone1)
We have maintained a single repo for all the docker images. If there is confusion we can go ahead and create seperate docker repos too. Do let me know if that helps and answers your query.

**Comment by takahser** [comment]
@varun24143 ok, got it. No need to add a new repo, I was just confused initially. 👍
Alright, so I did checkout [Phase2_Milestone1](https://github.com/WowLabz/dot_marketplace_docker/tree/Phase2_Milestone1) and ran `docker-compose up --build -d`. I now have a container running on port 9944:

```bash
$ docker ps
CONTAINER ID   IMAGE                  COMMAND                  CREATED      STATUS          PORTS                                       NAMES
e57b44b66225   dot_marketplace_node   "/usr/local/bin/node…"   3 days ago   Up 15 seconds   0.0.0.0:9944->9944/tcp, :::9944->9944/tcp   dot_marketplace_docker_dot_marketplace_node_1
```

However, when I try to access the server on that port the request fails with a **HTTP 400 (Bad Request)** and the following error message:

> WebSocket Protocol Error: Unable to parse WebSocket key.

So I'm not sure if I'm missing something or how should to connect to it. Am I supposed  to connect to it through polkadotJs, for example? In general, I think it would be helpful if your [documentation](https://github.com/WowLabz/dot-marketplace-v2/blob/Phase2_Milestone1/testing_guide_phase2_milestone1.md) would add some information on that.

**Comment by varun24143** [comment]
> @varun24143 ok, got it. No need to add a new repo, I was just confused initially. +1 Alright, so I did checkout [Phase2_Milestone1](https://github.com/WowLabz/dot_marketplace_docker/tree/Phase2_Milestone1) and ran `docker-compose up --build -d`. I now have a container running on port 9944:
> 
> ```shell
> $ docker ps
> CONTAINER ID   IMAGE                  COMMAND                  CREATED      STATUS          PORTS                                       NAMES
> e57b44b66225   dot_marketplace_node   "/usr/local/bin/node…"   3 days ago   Up 15 seconds   0.0.0.0:9944->9944/tcp, :::9944->9944/tcp   dot_marketplace_docker_dot_marketplace_node_1
> ```
> 
> However, when I try to access the server on that port the request fails with a **HTTP 400 (Bad Request)** and the following error message:
> 
> > WebSocket Protocol Error: Unable to parse WebSocket key.
> 
> So I'm not sure if I'm missing something or how should to connect to it. Am I supposed to connect to it through polkadotJs, for example? In general, I think it would be helpful if your [documentation](https://github.com/WowLabz/dot-marketplace-v2/blob/Phase2_Milestone1/testing_guide_phase2_milestone1.md) would add some information on that.

> @takahser Thanks for testing it out. Yes, there was a rust dependency which was causing the error, we have made the change and pushed the latest commit. You can try and run it now, as mentioned in the testing guide. Once the build is complete then you can connect your local node to Polkadot JS UI to test out the functionality. Hope this helps

**Comment by takahser** [comment]
@varun24143 sorry for the delay here, I just wanted to let you know that I didn't forget about your delivery. Currently we have a large backlog, hence please accept my apologies for any further delay. I'll try to have a look either tomorrow or  early next week at this again. Thanks for your understanding. 🙏

**Comment by varun24143** [comment]
> @varun24143 sorry for the delay here, I just wanted to let you know that I didn't forget about your delivery. Currently we have a large backlog, hence please accept my apologies for any further delay. I'll try to have a look either tomorrow or early next week at this again. Thanks for your understanding. pray

> @takahser No issues at all. We totally understand that there would be lot of deliveries to keep an eye on. Once you are done do let us know. I just wanted to share that we have completed second milestone as well, would you suggest submitting that as well along with this or we wait for you to first review this. Do let me know.

**Comment by takahser** [comment]
Hey @varun24143
Thanks again for your patience and sorry for the delay here.
Currently the build is failing on my machine:

```bash
$ docker-compose up --build -d
Building dot_marketplace_node
Sending build context to Docker daemon   2.56kB
Step 1/14 : FROM paritytech/ci-linux:production AS builder
 ---> cabfeffb97bd
Step 2/14 : RUN apt-get update     && apt-get install -y git
 ---> Using cache
 ---> f8207d85dc30
Step 3/14 : RUN git clone --single-branch -b Phase2_Milestone1 https://github.com/WowLabz/dot-marketplace-v2.git
 ---> Using cache
 ---> 9b309a18171c
Step 4/14 : WORKDIR /builds/dot-marketplace-v2
 ---> Using cache
 ---> f429dbc2647a
Step 5/14 : RUN rustup update
 ---> Using cache
 ---> 644a65cdf71f
Step 6/14 : RUN rustup install 1.59.0
 ---> Using cache
 ---> d64c647a8ce4
Step 7/14 : RUN rustup default nightly
 ---> Using cache
 ---> 93a0331709bd
Step 8/14 : RUN cargo update -p parity-db
 ---> Using cache
 ---> b17570fc6af2
Step 9/14 : RUN cargo build --release
 ---> Running in 36447864ad3c
 Downloading crates ...
  Downloaded proc-macro-error-attr v1.0.4
  Downloaded rand_chacha v0.3.1
  Downloaded thiserror v1.0.30
  Downloaded tinyvec_macros v0.1.0
  Downloaded snow v0.8.0
  Downloaded approx v0.5.0
  Downloaded ansi_term v0.12.1
  Downloaded derive_more v0.99.16
  Downloaded ipnet v2.3.1
  Downloaded jsonrpc-core-client v18.0.0

  (...)
  
   Compiling cranelift-codegen v0.78.0
   Compiling frame-support-procedural-tools v4.0.0-dev (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
   Compiling jsonrpc-derive v18.0.0
   Compiling parity-scale-codec-derive v2.3.1
   Compiling scale-info-derive v1.0.0
   Compiling sp-runtime-interface-proc-macro v4.0.0-dev (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
   Compiling sp-api-proc-macro v4.0.0-dev (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
error: cannot find macro `asm` in this scope
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:57:5
   |
57 |     asm!(
   |     ^^^
   |
   = note: consider importing one of these items:
           std::arch::asm
           core::arch::asm

error: cannot find macro `asm` in this scope
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:12:5
   |
12 |     asm!(
   |     ^^^
   |
   = note: consider importing one of these items:
           std::arch::asm
           core::arch::asm

error: cannot find macro `asm` in this scope
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:26:5
   |
26 |     asm!(
   |     ^^^
   |
   = note: consider importing one of these items:
           std::arch::asm
           core::arch::asm

error: cannot find macro `asm` in this scope
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:44:5
   |
44 |     asm!(
   |     ^^^
   |
   = note: consider importing one of these items:
           std::arch::asm
           core::arch::asm

error: cannot find macro `asm` in this scope
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:73:5
   |
73 |     asm!(
   |     ^^^
   |
   = note: consider importing one of these items:
           std::arch::asm
           core::arch::asm

error: cannot find macro `asm` in this scope
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:93:5
   |
93 |     asm!(
   |     ^^^
   |
   = note: consider importing one of these items:
           std::arch::asm
           core::arch::asm

error: cannot find macro `asm` in this scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:114:5
    |
114 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:136:5
    |
136 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:159:5
    |
159 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:183:5
    |
183 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:208:5
    |
208 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:234:5
    |
234 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:261:5
    |
261 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:289:5
    |
289 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: could not compile `rsix` due to 14 previous errors
warning: build failed, waiting for other jobs to finish...
The command '/bin/sh -c cargo build --release' returned a non-zero code: 101
ERROR: Service 'dot_marketplace_node' failed to build : Build failed
```

I'm using a Ubuntu 20.04.4 LTS machine running `docker-compose version 1.29.2, build 5becea4c`.

Revision:
```bash
$ git log
commit ebbd573b956a1ca82372286ccce323a75b61e3aa (HEAD -> Phase2_Milestone1, origin/Phase2_Milestone1)
Author: loaki07 <loakesh.i@gmail.com>
Date:   Tue Apr 12 23:25:25 2022 +0530

    modify: dockerfile to use 1.58.0-nightly
```

>  I just wanted to share that we have completed second milestone as well, would you suggest submitting that as well along with this or we wait for you to first review this.

You can submit it already now, and I'll try to look at them together. 👍

**Comment by varun24143** [comment]
> @takahser We tested on all the machines and it is working fine, we have made one more change to the dockerfile to accommodate the nightly version with a specific tag now. That is working fine as well. Kindly review it again and let me know. I will also go ahead and submit the second milestone too.

**Comment by varun24143** [comment]
> > @takahser We tested on all the machines and it is working fine, we have made one more change to the dockerfile to accommodate the nightly version with a specific tag now. That is working fine as well. Kindly review it again and let me know. I will also go ahead and submit the second milestone too.

> @takahser Please do review it once you get free. Just an update we have submitted phase2_milestone2 as well. Please find it [here](https://github.com/w3f/Grant-Milestone-Delivery/pull/441)

**Comment by takahser** [comment]
@varun24143 thanks for pinging me, I'll have a look at it again early next week. Pls accept my apology for the delay and let me thank you for your patience. 🙏

**Comment by takahser** [comment]
@varun24143 I tried running docker-compose again, and it worked:

```bash
$ docker-compose up --build -d
Building dot_marketplace_node
Sending build context to Docker daemon  2.077GB
Step 1/4 : FROM paritytech/ci-linux:production
 ---> cabfeffb97bd
Step 2/4 : WORKDIR /var/www/dot-marketplace-v2
 ---> Running in 8d9cd1aaab51
Removing intermediate container 8d9cd1aaab51
 ---> 23b85c77a375
Step 3/4 : COPY . /var/www/dot-marketplace-v2
 ---> 9394f5fc090d
Step 4/4 : EXPOSE 9944
 ---> Running in 4f57e933fb98
Removing intermediate container 4f57e933fb98
 ---> 2e2e39de0993
Successfully built 2e2e39de0993
Successfully tagged dot_marketplace_node:latest
Creating dot-marketplace-v2_dot_marketplace_node_1 ... done

$ docker ps
CONTAINER ID   IMAGE                  COMMAND                  CREATED          STATUS          PORTS                                       NAMES
456af7b552f9   dot_marketplace_node   "bash -c 'cargo buil…"   45 seconds ago   Up 44 seconds   0.0.0.0:9944->9944/tcp, :::9944->9944/tcp   dot-marketplace-v2_dot_marketplace_node_1
```

However, when trying to [run](https://github.com/WowLabz/dot-marketplace-v2/blob/06ca3b17963a93d09a05a73e63ac5aebc813de10/README.md#run) manually, it still fails with the same error as I had with docker-compose earlier, before you deployed the fix:

```bash
$ cargo run --release -- --dev --tmp
    Blocking waiting for file lock on build directory
   Compiling zstd-sys v1.6.1+zstd.1.5.0
   Compiling zstd-safe v4.1.1+zstd.1.5.0
   Compiling statrs v0.15.0
   Compiling sp-panic-handler v4.0.0-dev (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
   Compiling zstd v0.9.0+zstd.1.5.0
   Compiling scale-info v1.0.0
   Compiling impl-codec v0.5.1
   Compiling sp-storage v4.0.0-dev (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
   Compiling sp-wasm-interface v4.0.0-dev (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
   Compiling sp-tracing v4.0.0-dev (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
   Compiling sp-version-proc-macro v4.0.0-dev (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
   Compiling cranelift-codegen v0.78.0
   Compiling libp2p-relay v0.4.0
   Compiling libp2p-gossipsub v0.33.0
   Compiling sc-network v0.10.0-dev (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
   Compiling fork-tree v3.0.0 (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
   Compiling multiaddr v0.13.0
   Compiling libp2p-core v0.30.0
   Compiling libp2p-kad v0.32.0
   Compiling libp2p-identify v0.31.0
   Compiling cid v0.6.1
   Compiling rsix v0.23.9
   Compiling hyper v0.14.14
   Compiling libp2p-rendezvous v0.1.0
   Compiling libp2p-floodsub v0.31.0
   Compiling libp2p-plaintext v0.30.0
   Compiling libp2p-noise v0.33.0
   Compiling librocksdb-sys v6.20.3
   Compiling jsonrpc-core-client v18.0.0
   Compiling sp-maybe-compressed-blob v4.1.0-dev (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
   Compiling rocksdb v0.17.0
error: cannot find macro `asm` in this scope
  --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:57:5
   |
57 |     asm!(
   |     ^^^
   |
   = note: consider importing one of these items:
           std::arch::asm
           core::arch::asm

error: cannot find macro `asm` in this scope
  --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:12:5
   |
12 |     asm!(
   |     ^^^
   |
   = note: consider importing one of these items:
           std::arch::asm
           core::arch::asm

   Compiling sp-externalities v0.10.0-dev (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
error: cannot find macro `asm` in this scope
  --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:26:5
   |
26 |     asm!(
   |     ^^^
   |
   = note: consider importing one of these items:
           std::arch::asm
           core::arch::asm

error: cannot find macro `asm` in this scope
  --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:44:5
   |
44 |     asm!(
   |     ^^^
   |
   = note: consider importing one of these items:
           std::arch::asm
           core::arch::asm

error: cannot find macro `asm` in this scope
  --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:73:5
   |
73 |     asm!(
   |     ^^^
   |
   = note: consider importing one of these items:
           std::arch::asm
           core::arch::asm

error: cannot find macro `asm` in this scope
  --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:93:5
   |
93 |     asm!(
   |     ^^^
   |
   = note: consider importing one of these items:
           std::arch::asm
           core::arch::asm

error: cannot find macro `asm` in this scope
   --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:114:5
    |
114 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:136:5
    |
136 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:159:5
    |
159 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:183:5
    |
183 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:208:5
    |
208 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:234:5
    |
234 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:261:5
    |
261 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

error: cannot find macro `asm` in this scope
   --> /home/seraya/.cargo/registry/src/github.com-1ecc6299db9ec823/rsix-0.23.9/src/imp/linux_raw/arch/inline/x86_64.rs:289:5
    |
289 |     asm!(
    |     ^^^
    |
    = note: consider importing one of these items:
            std::arch::asm
            core::arch::asm

   Compiling linregress v0.4.4
   Compiling substrate-wasm-builder v5.0.0-dev (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
   Compiling primitive-types v0.10.1
   Compiling sp-arithmetic v4.0.0-dev (https://github.com/paritytech/substrate.git?tag=monthly-2021-12#b6c1c1bc)
   Compiling frame-metadata v14.0.0
   Compiling finality-grandpa v0.14.4
error: could not compile `rsix` due to 14 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed
```

I also get the same errors when running `cargo build --release`. Could you deploy the fix there as well?

**Comment by varun24143** [comment]
> 

> @takahser Hi, Thanks for checking this out. The issue with manual run is prevailing because of Rust's latest version, which uses the updated version of `rsix` crate but with a different name `rustix`. Hence to fix it for the manual build you will have to downgrade the rust version to `1.58.0-nightly`. We have updated the Readme's accordingly for manual build as well. Please refer once.

Note : The current build is supported by Rust Nightly `1.58.0-nightly` due to its dependency on a specific crate used in this codebase.
Please update the rust version accordingly to run the build.

```sh
rustup update

# installing rust version 1.58.0-nightly
rustup install nightly-2021-11-01

# setting rust default version as 1.58.0-nightly
rustup default nightly-2021-11-01

# run the node
cargo run --release -- --dev --tmp
```

After this it will work with manual setup too. Do let me know if it still throws any errors.

**Comment by takahser** [comment]
@varun24143 Despite running the commands:

image

However, it still fails:

```bash
$ cargo run --release -- --dev --tmp

(...)

error: failed to run custom build command for `node-template-runtime v4.0.0-dev (/home/xxx/repos/dot-marketplace-v2/runtime)`

Caused by:
  process didn't exit successfully: `/home/xxx/repos/dot-marketplace-v2/target/release/build/node-template-runtime-3d8f729c9dd239cd/build-script-build` (exit status: 1)
  --- stderr
  Rust WASM toolchain not installed, please install it!

  Further error information:
  ------------------------------------------------------------
     Compiling wasm-test v1.0.0 (/tmp/.tmpboCZmJ)
  error[E0463]: can't find crate for `std`
    |
    = note: the `wasm32-unknown-unknown` target may not be installed
    = help: consider downloading the target with `rustup target add wasm32-unknown-unknown`
    = help: consider building the standard library from source with `cargo build -Zbuild-std`

  For more information about this error, try `rustc --explain E0463`.
  error: could not compile `wasm-test` due to previous error
  warning: build failed, waiting for other jobs to finish...
  error: build failed
  ------------------------------------------------------------

warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Review by takahser — CHANGES_REQUESTED** [review]
See my previous comment.

**Comment by Loaki07** [comment]
> @varun24143 Despite running the commands...
> 
> image
> 
> ... the build still fails:
> 
> ```shell
> $ cargo run --release -- --dev --tmp
> 
> (...)
> 
> error: failed to run custom build command for `node-template-runtime v4.0.0-dev (/home/xxx/repos/dot-marketplace-v2/runtime)`
> 
> Caused by:
>   process didn't exit successfully: `/home/xxx/repos/dot-marketplace-v2/target/release/build/node-template-runtime-3d8f729c9dd239cd/build-script-build` (exit status: 1)
>   --- stderr
>   Rust WASM toolchain not installed, please install it!
> 
>   Further error information:
>   ------------------------------------------------------------
>      Compiling wasm-test v1.0.0 (/tmp/.tmpboCZmJ)
>   error[E0463]: can't find crate for `std`
>     |
>     = note: the `wasm32-unknown-unknown` target may not be installed
>     = help: consider downloading the target with `rustup target add wasm32-unknown-unknown`
>     = help: consider building the standard library from source with `cargo build -Zbuild-std`
> 
>   For more information about this error, try `rustc --explain E0463`.
>   error: could not compile `wasm-test` due to previous error
>   warning: build failed, waiting for other jobs to finish...
>   error: build failed
>   ------------------------------------------------------------
> 
> warning: build failed, waiting for other jobs to finish...
> error: build failed
> ```
> 
> Edit:
> 
> * I installed `wasm32-unknown-unknown` (yes, that's [actually the name of this library](https://github.com/rustwasm/wasm-bindgen/issues/979) zany_face)
> * Build is still failing though:
>   ```shell
>   $ cargo run --release -- --dev --tmp
>   
>   (...)
>   
>   error: failed to run custom build command for `node-template-runtime v4.0.0-dev (/home/xxx/repos/dot-marketplace-v2/runtime)`
>   
>   Caused by:
>     process didn't exit successfully: `/home/xxx/repos/dot-marketplace-v2/target/release/build/node-template-runtime-3d8f729c9dd239cd/build-script-build` (exit status: 101)
>     --- stderr
>     thread 'main' panicked at 'Writing `/home/xxx/repos/dot-marketplace-v2/target/release/wbuild/node-template-runtime/Cargo.toml` can not fail!', /home/xxx/.cargo/git/checkouts/substrate-7e08433d4c370a21/b6c1c1b/utils/wasm-builder/src/lib.rs:145:33
>     note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
>   warning: build failed, waiting for other jobs to finish...
>   error: build failed
>   ```

@takahser looks like you needed to install `wasm32-unknown-unknown`. Once you have set default rust version to `nightly-2021-11-01` and the required targets your output for command `rustup show` should look like the below:

rust_config

Could you please post/confirm that these are matching the output on your system.
Here is the [installation guide](https://docs.substrate.io/v3/getting-started/installation/) for substrate/rust for reference.

Suggestions:
1. Once the config is set/updated, please run `cargo clean` in the project directory, to remove the older builds.
2. Then you can proceed with the regular build commands
```shell
cargo run --release -- --dev --tmp

# or

cargo build --release

./target/release/node-template --dev --tmp

```

**Review by takahser — APPROVED** [review]
@varun24143 thanks for your patience.
Everything is fine now, you can find my updated evaluation [here](https://github.com/w3f/Grant-Milestone-Delivery/blob/1f27e6caea6017ef24de88150b30aad0ecb94d46/evaluations/dot-marketplace-v2_1_takahser.md).

**Pull request merged**